python/pyarrow/_dataset_parquet.pyx

```diff
@@ -769,6 +777,14 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
     def pre_buffer(self, bint pre_buffer):
         self.arrow_reader_properties().set_pre_buffer(pre_buffer)
 
+    @property
+    def cache_options(self):
+        return CacheOptions.wrap(self.arrow_reader_properties().cache_options())
+
+    @cache_options.setter
+    def cache_options(self, CacheOptions options):
+        self.arrow_reader_properties().set_cache_options(options.unwrap())
```

Comment:
When adding this new parameter/property here, you will probably also have to update `equals` and `__reduce__` a bit below.

Reply:
Yes, should be fixed now. I'm trying to remember if I missed that originally or if I lost it in my rebase :sweat_smile: